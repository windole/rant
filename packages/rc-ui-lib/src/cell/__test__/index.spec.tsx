import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Cell from '..';

describe('Cell', () => {
  let wrapper;

  afterEach(() => {
    wrapper.unmount();
    jest.restoreAllMocks();
  });

  it('should render value slot correctly', () => {
    wrapper = mount(<Cell value="Custom Value" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render title slot correctly', () => {
    wrapper = mount(<Cell title="Custom Title" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render label slot correctly', () => {
    wrapper = mount(<Cell title="Title" label="Custom Label" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render icon slot correctly', () => {
    wrapper = mount(<Cell icon="location-o" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render extra slot correctly', () => {
    wrapper = mount(<Cell title="Title" value="Value" extra="Custom Extra" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should change arrow direction when using arrow-direction prop', () => {
    wrapper = mount(<Cell title="Title" value="Value" isLink arrowDirection="down" />);
    expect(wrapper.find('.rc-cell__right-icon').exists()).toBeTruthy();
    expect(wrapper.find('.van-icon-arrow-down').exists()).toBeTruthy();
  });

  it('should change title style when using title-style prop', () => {
    wrapper = mount(<Cell title="Title" titleStyle={{ color: 'red' }} />);
    const title = wrapper.find('.rc-cell__title');
    expect(title.getDOMNode().style.color).toEqual('red');
  });

  it('should change icon class prefix when using icon-prefix prop', () => {
    wrapper = mount(<Cell title="Title" icon="success" iconPrefix="my-icon" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should allow to disable clicakble when using is-link prop', () => {
    wrapper = mount(<Cell title="Title" isLink clickable={false} />);
    expect(wrapper.find('.rc-cell--clickable').exists()).toBeTruthy();
  });

  it('should render correctly when wrapperd with Cell.Group', () => {
    wrapper = mount(
      <Cell.Group>
        <Cell title="单元格" value="内容" />
        <Cell title="单元格" value="内容" label="描述信息" />
      </Cell.Group>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly when using title prop in Cell.Group', () => {
    wrapper = mount(
      <div>
        <Cell.Group title="分组1">
          <Cell title="单元格" value="内容" />
        </Cell.Group>
        <Cell.Group title="分组2">
          <Cell title="单元格" value="内容" />
        </Cell.Group>
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly when using inset prop in Cell.Group', () => {
    wrapper = mount(
      <Cell.Group inset>
        <Cell title="单元格" value="内容" />
        <Cell title="单元格" value="内容" label="描述信息" />
      </Cell.Group>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
