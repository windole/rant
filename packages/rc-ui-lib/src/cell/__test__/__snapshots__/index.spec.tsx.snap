// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell should change icon class prefix when using icon-prefix prop 1`] = `"<div class=\\"rc-cell\\"><i class=\\"rc-badge__wrapper rc-cell__left-icon my-icon my-icon-success\\"></i><div class=\\"rc-cell__title\\">Title</div></div>"`;

exports[`Cell should render correctly when using inset prop in Cell.Group 1`] = `undefined`;

exports[`Cell should render correctly when using title prop in Cell.Group 1`] = `undefined`;

exports[`Cell should render correctly when wrapperd with Cell.Group 1`] = `undefined`;

exports[`Cell should render correctly when wrapperd with Cell.Group 2`] = `undefined`;

exports[`Cell should render extra slot correctly 1`] = `"<div class=\\"rc-cell\\"><div class=\\"rc-cell__title\\">Title</div><div class=\\"rc-cell__value\\"><span>Value</span></div>Custom Extra</div>"`;

exports[`Cell should render icon slot correctly 1`] = `
<Cell
  icon="location-o"
>
  <div
    className="rc-cell"
  >
    <Icon
      className="rc-cell__left-icon"
      name="location-o"
      tag="i"
    >
      <Badge
        className="rc-cell__left-icon van-icon van-icon-location-o"
        showZero={true}
        style={
          Object {
            "color": undefined,
            "fontSize": undefined,
          }
        }
        tag="i"
      >
        <i
          className="rc-badge__wrapper rc-cell__left-icon van-icon van-icon-location-o"
          style={
            Object {
              "color": undefined,
              "fontSize": undefined,
            }
          }
        />
      </Badge>
    </Icon>
  </div>
</Cell>
`;

exports[`Cell should render label slot correctly 1`] = `
<Cell
  label="Custom Label"
  title="Title"
>
  <div
    className="rc-cell"
  >
    <div
      className="rc-cell__title"
    >
      Title
      <div
        className="rc-cell__label"
      >
        Custom Label
      </div>
    </div>
  </div>
</Cell>
`;

exports[`Cell should render title slot correctly 1`] = `
<Cell
  title="Custom Title"
>
  <div
    className="rc-cell"
  >
    <div
      className="rc-cell__title"
    >
      Custom Title
    </div>
  </div>
</Cell>
`;

exports[`Cell should render value slot correctly 1`] = `
<Cell
  value="Custom Value"
>
  <div
    className="rc-cell"
  >
    <div
      className="rc-cell__value rc-cell__value--alone"
    >
      <span>
        Custom Value
      </span>
    </div>
  </div>
</Cell>
`;
